### PR TITLE
enable unit tests (rebased)

### PR DESCRIPTION
**Pull request by balcy**
rebased https://github.com/ubports/morph-browser/pull/433 and removed two test cases that are no longer applicable.

### CODE REVIEW

**Comment by UniversalSuperBox** [comment]
While I'm unable to test this right now, I move to say that we do not merge any new changes to morph-browser until this is successfully merged. That way we won't break the tests before they're even enabled again. How does that sound?

**Comment by balcy** [comment]
sounds good to me :)

File: tests/unittests/history-model/tst_HistoryModelTests.cpp
Comment (UniversalSuperBox) [inline tests/unittests/history-model/tst_HistoryModelTests.cpp]: TRY_COMPARE should wait for the change by itself, so qWait should not be necessary... is it not waiting long enough?

File: tests/unittests/qml/tst_ChromeController.qml
Comment (UniversalSuperBox) [inline tests/unittests/qml/tst_ChromeController.qml]: How did you get accented M's?

File: tests/unittests/qml/tst_ChromeController.qml
Comment (UniversalSuperBox) [inline tests/unittests/qml/tst_ChromeController.qml]: console.log left in accidentally?

File: tests/unittests/qml/tst_MorphWebView.qml
Comment (UniversalSuperBox) [inline tests/unittests/qml/tst_MorphWebView.qml]: Either `skip()` it or remove it. No use in commenting it.

File: tests/unittests/qml/tst_PreviewManager.qml
Comment (UniversalSuperBox) [inline tests/unittests/qml/tst_PreviewManager.qml]: Accidentally left in console.log?

File: tests/unittests/qml/tst_WebProcessMonitor.qml
Comment (UniversalSuperBox) [inline tests/unittests/qml/tst_WebProcessMonitor.qml]: Why was this moved from tryCompare to a wait followed by a compare?

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]
Looking good! Let's just clean it up a bit.

File: tests/unittests/qml/tst_WebProcessMonitor.qml
Comment (balcy) [reply tests/unittests/qml/tst_WebProcessMonitor.qml]: I think it was because the rest of the test_killed() method uses compare().
It was a long time ago, therefore I don't exactly remember, but I think there was an error when running it.

File: tests/unittests/qml/tst_PreviewManager.qml
Comment (balcy) [reply tests/unittests/qml/tst_PreviewManager.qml]: I guess so, it was for debugging -> I'll remove it

File: tests/unittests/qml/tst_ChromeController.qml
Comment (balcy) [reply tests/unittests/qml/tst_ChromeController.qml]: I guess so, it was for debugging -> I'll remove it

File: tests/unittests/qml/tst_ChromeController.qml
Comment (balcy) [reply tests/unittests/qml/tst_ChromeController.qml]: my keyboard is not short of surprises : )

File: tests/unittests/history-model/tst_HistoryModelTests.cpp
Comment (balcy) [reply tests/unittests/history-model/tst_HistoryModelTests.cpp]: If I remember correctly there were timing related errors without the wait, but I don't find the build for it.
This was the original commit for it: https://github.com/ubports/morph-browser/commit/a0846f66671914314ee33bd4001d64d18c95260f#diff-561e4157ce1ef4f964f8183110ea6e4bb19b6fa0310e4c1831d931a7aec7cbae

File: tests/unittests/qml/tst_WebProcessMonitor.qml
Comment (balcy) [reply tests/unittests/qml/tst_WebProcessMonitor.qml]: ok it seems to build using the old way, too -> restoring it for now

File: tests/unittests/history-model/tst_HistoryModelTests.cpp
Comment (balcy) [reply tests/unittests/history-model/tst_HistoryModelTests.cpp]: ok if it builds without it let's remove it for now. Maybe it just happened back then / with the old Qt version.

**Review by UniversalSuperBox — APPROVED** [review]
Alright, changes look good. I've given it a try and haven't found any new warnings or any other indicators of bad behavior. Let's go for wider testing!

**Pull request merged**
